src/stat/hist.ts

```diff
@@ -0,0 +1,51 @@
+import { Tensor1D } from "@tensorflow/tfjs-core";
+import { checkJSArray } from "../utils/validation";
+
+/**
+ * Get histogram data for given input
+ * @param xData inpur data of shape [nSamples, ]
+ * @param params parameters
+ * Option in params
+ * ----
+ * `bins`: number of cells, **default=50**
+ * `leftLim`: left limit to calculate, **default=min value of input data**
+ * `rightLim`: right limit to calculate, **default=max value of input data**
+ * @returns
+ * { steps; number[], counts: number[] }
+ * `steps`: array of giving the breakpoints between histogram cells,
+ * `counts`: array of number of data points in histogram cells
+ */
+export const getHistData = (xData: Tensor1D | number[], params?: {
+    bins?: number,
+    leftLim?: number,
+    rightLim?: number
+  }): {
+    steps: number[],
+    counts: number[],
+  } => {
+  const xArray = checkJSArray(xData, 'float32', 1) as number[];
+  const nData = xArray.length;
```

Comment:
Why the name of an array length is called `*data`?

Reply:
number of data

Reply:
How about the complete `numberOfData`?